pkgs/by-name/th/thorium/package.nix

```
@@ -0,0 +1,46 @@
+{
+  stdenv,
+  fetchurl,
+  undmg,
+  lib,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "thorium";
+  version = "126.0.6478.231";
+
+  src = fetchurl {
+    url = "https://github.com/Alex313031/Thorium-MacOS/releases/download/M${finalAttrs.version}/Thorium_MacOS_ARM64.dmg";
+    hash = "sha256-BBqkNbQ7QjCMfU8yQkiRqV3g9ipjco1XKxR7VYVWhig=";
+  };
+
+  nativeBuildInputs = [ undmg ];
+
+  unpackPhase = ''
+    runHook preUnpack
+    undmg "$src"
+    rm ./Applications
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/Applications
+    cp -r ./Thorium.app $out/Applications
+    runHook postInstall
```

This creates a relative symbolic link "$out/bin/thorium" pointing to "../Applications/Thorium.app/Contents/MacOS/Thorium". The use of `-r` and the absolute target path is for readability.

```suggestion
    mkdir -p "$out/bin"
    ln -sr "$out/Applications/Thorium.app/Contents/MacOS/Thorium" "$out/bin/thorium"
    runHook postInstall
```

Now, it would be a great idea to specify the `meta.mainProgram = "thorium";`.